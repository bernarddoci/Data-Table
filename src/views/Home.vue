<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" class="logo" />
    <h1>Engineering Application</h1>
    <button @click="this.showTable" to="/data-table" class="show-table-btn">
      Show Table >
    </button>
  </div>
</template>

<script>
export default {
  name: "home",
  methods: {
    showTable() {
      this.$router.push({ path: "/data-table" });
    }
  }
};
</script>

<style lang="scss" scoped>
.home {
  margin-top: 10%;
  .logo {
    width: 30%;
  }
  .show-table-btn {
    width: 200px;
    height: 50px;
    border: none;
    background-color: #0191a9;
    color: #fff;
    font-size: 1rem;
    font-weight: 700;
    cursor: pointer;
  }
}
@media only screen and (max-width: 575px) {
  .home {
    margin-top: 20%;
    .logo {
      width: 80%;
    }
  }
}
</style>
