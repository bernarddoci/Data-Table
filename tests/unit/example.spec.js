import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home.vue";

describe("Home.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.element).toMatchSnapshot();
  });
});
