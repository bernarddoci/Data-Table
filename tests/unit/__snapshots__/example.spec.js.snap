// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders correctly 1`] = `
<div
  class="home"
>
  <img
    alt="Vue logo"
    class="logo"
    src="../assets/logo.png"
  />
   
  <h1>
    Engineering Application
  </h1>
   
  <button
    class="show-table-btn"
    to="/data-table"
  >
    
    Show Table &gt;
  
  </button>
</div>
`;
